piqueserver/config.py

```diff
@@ -0,0 +1,86 @@
+import toml
+from pprint import pprint
+
+class ConfigException(Exception):
+    pass
+
+
+class ConfigEngine():
+    '''
+    configuration engine that handles loading and saving config,
+    and all the things
+    '''
+    def __init__(self):
+        self.raw_config = {}
+        self.style = 'TOML'
+
+    def load_config(self, config_file, style='TOML'):
+        # TODO: could also have this merge into current config if available
+        self.style = style
+
+        if style == 'TOML':
+            self.raw_config = toml.load(open(config_file))
+        else:
+            raise Exception()
+
+    def dump_config(self, out_file=None, style='TOML'):
+        if out_file is None:
+            pprint(self.raw_config)
+        # TODO
+        pass
+
+    def get(self, name, default):
+        # TODO: handle nested config values
+        # XXX: should requesting a config value that is a dictionary object
```

Comment:
I believe not just returning the dictionary object will only cause arbitrary and weird restrictions in places.